cluster-controller/src/main/java/io/enmasse/barnabas/controller/cluster/operations/UpdateZookeeperClusterOperation.java

```diff
@@ -38,7 +38,7 @@ public void execute(Vertx vertx, K8SUtils k8s, Handler<AsyncResult<Void>> handle
 
                 if (zkConfigMap != null)    {
                     zk = ZookeeperCluster.fromConfigMap(zkConfigMap);
-                    diff = zk.diff(k8s.getStatefulSet(namespace, name + "-zookeeper"));
+                    diff = zk.diff(k8s.getStatefulSet(namespace, name + "-zookeeper"), k8s.getConfigmap(namespace, name + "-zookeeper-metrics-config"));
```

Comment:
I think it would be great if we can find a way how to avoid this `+ "-zookeeper"` and `+ "-zookeeper-metrics-config"`. Wat about having some static methods in the `ZookeeperCluster` class or something like that? _(Sorry, I know I introduced this :-()_

Reply:
The `ZookeeperCluster` class has a `getMetricsConfigName` name which provides this information but in this case it could happen that the updated cm doesn't have the metrics configuration anymore so the related metrics config name is a null String. I could check that and passing a null configMap avoind the exception from `k8s.getConfigmap` about a null name. Or did you mean something different ?

Reply:
What I meant is something such as a method `ZookeeperCluster.getMetricsConfigName(String name)` which would return `name + "-zookeeper-metrics-config"`. This method can be static and can be used in these calls. 

I.e. you would replace this:
`diff = zk.diff(k8s.getStatefulSet(namespace, name + "-zookeeper"), k8s.getConfigmap(namespace, name + "-zookeeper-metrics-config"));`
with something like this:
`diff = zk.diff(k8s.getStatefulSet(namespace, name + "-zookeeper"), k8s.getConfigmap(namespace, ZookeeperCluster.getMetricsConfigName(name)));`